src/utils.cpp

```diff
@@ -507,6 +507,13 @@ std::vector<int> get_possible_cpus()
   return read_cpu_range("/sys/devices/system/cpu/possible");
 }
 
+int get_max_cpu_id()
+{
+  const auto cpus = get_possible_cpus();
+  auto it = std::max_element(cpus.begin(), cpus.end());
+  return it != cpus.end() ? *it : 0;
```

Comment:
Is this check necessary, if we can't get any possible cpus then I don't know how bpftrace itself is running :)

Reply:
Yea not really :/ We could probably just throw a FatalUserException instead

Reply:
I would argue you can just blindly do the dereference. We could add something more friendly if we get reports that bpftrace is actually dying here.

Reply:
Fixed in latest PR update (coming soon)